qir/qat/AdaptorFactory/ConfigurableQirAdaptorFactory.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation.
```

Comment:
The corresponding header file `ConfigurableQirAdaptorFactory.hpp` has been renamed/moved (from `ConfigurableProfileGenerator.hpp`) and edited. That is why the history is preserved and the diff for the updated lines only is shown.

I suspect that this .cpp file has also been renamed/moved and edited. But it is shown as newly created (and the `ConfigurableProfileGenerator.cpp` is shown as removed). Likely you renamed/moved _manually_ instead of renaming/moving with `git mv ....`. Would be nice to preserve teh history and see the diff for the changed lines only. Consider re-doing the renaming/move by means of `git mv ....`.
(I don't insist, up to you)

Reply:
This clearly should have used git mv. I suspect the problem is that I renamed the file in the editor at the same time as I did substantial changes.

Do you know if it is possible to fix this retrospectively?

Reply:
After reading about it, I am not sure that this is fixable: https://www.reddit.com/r/git/comments/nhnjtj/git_mv_deletes_all_previous_commit_history/

I will wait with merging to hear your suggestion.

Reply:
Looks like for the header file it worked.

Here is what comes to my mind: 
Checkout `main`, create a `tmp` branch on top of main, try to do `git mv ....` (for the .cpp file of interest) and commit. Then do the edits if needed (for the .cpp file of interest), and commit (from my previous experience I got an impression that it is always a good idea to separate the `git mv` from the edits into separate commits). Thus you get 2 more commits in the history of `tmp` branch. 
Then see the diff of the 2 last commits: `git log --grpah HEAD~2 HEAD > 01.diff`. If the history of the moved/renamed file is preserved (and the diff shows the edits only, and indicates that the file was moved/renamed) then I would do the following.

(The retention of the file history may be different for the 
file move case, 
file renaming case, 
move-and-renaming (in a single commit) case, 
each followed by edits. If time permits I would try to investigate and compare each)

Checkout the PR branch. Get the diff for it `git diff main HEAD > 02.diff`. From that diff file "02.diff" manually remove everything related to the file of interest (removal of the old file, addition of the new file). 
Then checkout the `tmp` branch again (having 2 commits discussed above), and apply that patch "02.diff" (in Linux/WSL something like `patch -p1 < 02.diff`). Commit to the `tmp` branch. Thus the tmp branch contains the 2 commits with manual `git mv` and edits for the file of interest, followed by all the changes except for that file of interest.
Then locally remove the PR branch `git branch -D <PR_branch>` and create again on the tip of `tmp` branch, `git checkout -b <PR_branch>`. Force-push that updated PR branch to git hub `git push -f origin <PR_branch>`.